client/js/components/show-hide.js

```diff
@@ -0,0 +1,42 @@
+const showHide = (state) => {
+  const el = state.el;
+  const activeClass = state.activeClass || 'is-active';
+  const trigger = el.querySelector('.js-show-hide-trigger');
+  const drawer = el.querySelector('.js-show-hide-drawer');
+
+  const init = () => {
+    el.setAttribute('aria-haspopup', 'true');
+    trigger.setAttribute('aria-expanded', getActive());
+  };
+
+  const setActive = (value) => {
+    if (value) {
+      el.classList.add(activeClass);
+      trigger.setAttribute('aria-expanded', 'true');
+    } else {
+      el.classList.remove(activeClass);
+      trigger.setAttribute('aria-expanded', 'false');
+    }
+  };
+
+  const getActive = () => {
+    return el.classList.contains(activeClass);
```

Comment:
Storing state in the DOM will be expensive as you'll need to read and write to is all the time - we should really look at using redux potentially.